include/revng/Yield/Support/GraphLayout/SugiyamaStyle/Compute.h

```diff
@@ -71,8 +72,47 @@ struct Configuration {
   layout::Dimension EdgeMarginSize;
 };
 
+namespace detail {
+
+bool computeImpl(InternalGraph &Internal, const Configuration &Configuration);
+
+} // namespace detail
+
+template<layout::HasLayoutableGraphTraits GraphType>
+inline bool
+computeInPlace(GraphType &&Graph, const Configuration &Configuration) {
+  using IG = InternalGraph;
+  auto [Internal, InputNodeLookup] = IG::make(std::forward<GraphType>(Graph));
+  if (!detail::computeImpl(Internal, Configuration))
+    return false;
+
+  Internal.template exportInto<GraphType>(InputNodeLookup);
+  return true;
+}
+
 /// A custom graph layering algorithm designed for pre-calculating majority of
 /// the expensive stuff needed for graph rendering.
-bool compute(Graph &Graph, const Configuration &Configuration);
+///
+/// \tparam Node The type of the data attached to each graph node
+/// \tparam Edge The type of the data attached to each graph edge
+///
+/// \param Graph An input graph
+/// \param Configuration An object configuring the specifics of the layout
+///
+/// \return The laid out version of the graph corresponding to \ref Graph
+template<typename Node, typename Edge = Empty>
+inline std::optional<layout::OutputGraph<Node, Edge>>
+compute(const layout::InputGraph<Node, Edge> &Graph,
+        const Configuration &Configuration) {
+  // TODO: rename into `compute` once `llvm::GraphTraits` has a concept support
+  //       for checking whether it's defined for a given type or not.
```

Comment:
This TODO seems outdated

Reply:
Ah, whoops, I forgot I switched the names around. This TODO is for `computeInPlace`.

Reply:
Done.